nornir/plugins/inventory/simple.py

```diff
@@ -15,28 +20,34 @@ def __init__(
         host_file: str = "hosts.yaml",
         group_file: str = "groups.yaml",
         defaults_file: str = "defaults.yaml",
+        hosts: Optional[HostsDict] = None,
+        groups: Optional[GroupsDict] = None,
+        defaults: Optional[VarsDict] = None,
         *args: Any,
         **kwargs: Any
     ) -> None:
-        yml = ruamel.yaml.YAML(typ="safe")
-        with open(host_file, "r") as f:
-            hosts = yml.load(f)
+        if hosts is None:
+            yml = ruamel.yaml.YAML(typ="safe")
+            with open(host_file, "r") as f:
+                hosts = yml.load(f)
 
-        groups: GroupsDict = {}
-        if group_file:
-            if os.path.exists(group_file):
-                with open(group_file, "r") as f:
-                    groups = yml.load(f) or {}
-            else:
-                logger.debug("File %r was not found", group_file)
-                groups = {}
+        if groups is None:
+            groups = {}
+            if group_file:
```

Comment:
Won't this and the defaults_file always be set to groups.yaml or defaults.yaml?

Reply:
Not necessarily, someone could pass an empty string or None.